dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPIImpl.java

```diff
@@ -0,0 +1,525 @@
+                + File.separator + "server" + File.separator;
+        final String dirPath = Config
+                .getStringProperty(SERVICE_INTEGRATION_DIR_PATH_KEY, supplier.get());
+        return Paths.get(dirPath).normalize().toString();
+    }
+
+    private Set<String> listAvailableYamlFiles() throws IOException {
+        final String ymlFilesPath = getServiceDescriptorDirectory();
+        final File basePath = new File(ymlFilesPath);
+        if (!basePath.exists()) {
+            basePath.mkdir();
+        }
+        Logger.debug(ServiceIntegrationAPIImpl.class,
+                () -> " ymlFiles are set under:  " + ymlFilesPath);
+        final Set<String> files = listFiles(ymlFilesPath);
+        if (!UtilMethods.isSet(files)) {
+            // No files were found a default empty descriptor will be created.
+            // ymlMapper.writeValue(new File(basePath, "sample-service-descriptor.yml"), emptyDescriptor());
+            // return listFiles(ymlFilesPath);
+           return Collections.emptySet();
+        } else {
+            return files;
+        }
+    }
+
+    private Set<String> listFiles(final String dir) throws IOException {
+        final Set<String> fileList = new HashSet<>();
+        try (final DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(dir))) {
+            for (final Path path : stream) {
+                if (!Files.isDirectory(path)) {
+                    final String fileName = path.getFileName().toString();
+                    if (fileName.endsWith("yaml") || fileName.endsWith("yml")) {
+                        fileList.add(path.toString());
+                    }
+                }
+            }
+        }
+        return fileList;
+    }
+
+    private List<ServiceDescriptorMeta> loadServiceDescriptors() throws IOException {
+        final ImmutableList.Builder<ServiceDescriptorMeta> builder = new ImmutableList.Builder<>();
+        final Set<String> fileNames = listAvailableYamlFiles();
+        for (final String fileName : fileNames) {
+            try {
+                final File file = new File(fileName);
+                final ServiceDescriptor serviceDescriptor = ymlMapper.readValue(file, ServiceDescriptor.class);
+                builder.add(new ServiceDescriptorMeta(serviceDescriptor, file.getName()));
+            } catch (IOException e) {
+                Logger.error(ServiceIntegrationAPIImpl.class,
+                        String.format("Error reading yml file `%s`.", fileName), e);
+            }
+        }
+
+        return builder.build();
+    }
+
+   private boolean validateServiceDescriptor(final ServiceDescriptor serviceDescriptor, final User user)
+           throws DotDataException, DotSecurityException {
+       if(UtilMethods.isNotSet(serviceDescriptor.getKey())){
+          throw new DotDataException("The required field `key` isn't set on the incoming file.");
+       }
+
+       if(serviceDescriptor.getKey().length() > 100){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)